### PR TITLE
feat!: support workflow_run event and use node 16

### PR DESCRIPTION
**Pull request by danez**
#### Summary

The changes:

- Use node16 runner (breaking). This happened because I wanted to use optional-chaining :)
- Support the `workflow_run` event. See https://github.com/netlify/cli/pull/4774
I tested this here: https://github.com/danez/cli/commit/0d048404ac9459624beb5e126de95e35ac8826f4#commitcomment-77740090
and here
https://github.com/danez/cli/pull/2
- Add more debug logging

### CODE REVIEW

File: src/lib/inputs.js [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
+import { context } from '@actions/github'
+
+const getPrNumber = () => {
+  const { eventName, payload } = context
+
+  if (eventName === 'pull_request') {
+    return payload.number
+  }
+
+  if (eventName === 'workflow_run' && payload?.workflow_run?.pull_requests?.length === 1) {
```
Comment (danez) [inline src/lib/inputs.js:13]: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_run

File: src/index.js [hunk 2]
```diff
@@ -67,7 +67,6 @@ const processPullRequest = async ({ headMetrics, job, octokit, owner, prNumber,
 const run = async function () {
   const { baseBranch, commitSha, job, owner, prNumber, ref, repo, rootPath, title, token } = getInputs()
   const headMetrics = await readDeltaFiles(rootPath)
-  const isHeadBranch = ref === `refs/heads/${baseBranch}`
```
Comment (danez) [inline src/index.js:70]: I inlined this directly in the `if` further down, to save one statement because the maximum is 15 according to eslint.

Comment (Skn0tt) [reply src/index.js:70]: :old-man-yells-at-eslint:

if this is the result of that eslint rule, then we should disbale it.

**Review by Skn0tt — APPROVED** [review]
Looks great!

**Pull request merged**